core/src/main/java/org/ehcache/Ehcache.java

```diff
@@ -1379,9 +1383,13 @@ public boolean remove(K key) {
 
     @Override
     public void removeAll() {
-      for (Iterator<Cache.Entry<K, V>> iter = new CacheEntryIterator(true); iter.hasNext(); ) {
-        K key = iter.next().getKey();
-        remove(key);
+      try {
+        for (Iterator<Entry<K, V>> iter = new CacheEntryIterator(true); iter.hasNext(); ) {
+          K key = iter.next().getKey();
+          remove(key);
+        }
+      } catch (CacheAccessException e) {
+        throw new RuntimeException(e);
```

Comment:
Throwing RuntimeException to be consistent with the behaviour of the other methods. If we decide to handle these too with resilience strategy, that will be done as a separate commit.